File: testcases/vertex_ai/geometry.test_3c3c033f.js
const {
    calculateCircleArea,
    calculateRectangleArea,
    calculateTriangleArea,
    calculateCirclePerimeter,
    calculateRectanglePerimeter,
    calculateDistance
} = require('../../geometry');

describe('calculateCircleArea', () => {
    /** */
     // Test case: Calculates the area of a circle with a positive radius.
     // /
    test('should calculate the area of a circle with a positive radius', () => {
        expect(calculateCircleArea(5)).toBeCloseTo(Math.PI * 5 * 5);
    });

    /** */
     // Test case: Calculates the area of a circle with a radius of 0.
     // /
    test('should calculate the area of a circle with a radius of 0', () => {
        expect(calculateCircleArea(0)).toBe(0);
    });

    /** */
     // Test case: Throws an error when the radius is negative.
     // /
    test('should throw an error when the radius is negative', () => {
        expect(() => calculateCircleArea(-5)).toThrow("Radius cannot be negative");
    });

    /** */
     // Test case: Throws an error when the radius is not a number.
     // /
    test('should throw an error when the radius is not a number', () => {
        expect(() => calculateCircleArea('abc')).toThrow("Radius must be a number");
    });

    /** */
     // Test case: Throws an error when the radius is NaN.
     // /
    test('should throw an error when the radius is NaN', () => {
        expect(() => calculateCircleArea(NaN)).toThrow("Radius must be a number");
    });

    /** */
     // Test case: Calculates the area of a circle with a large radius.
     // /
    test('should calculate the area of a circle with a large radius', () => {
        expect(calculateCircleArea(1000)).toBeCloseTo(Math.PI * 1000 * 1000);
    });

    /** */
     // Test case: Calculates the area of a circle with a decimal radius.
     // /
    test('should calculate the area of a circle with a decimal radius', () => {
        expect(calculateCircleArea(2.5)).toBeCloseTo(Math.PI * 2.5 * 2.5);
    });
});

describe('calculateRectangleArea', () => {
    /** */
     // Test case: Calculates the area of a rectangle with positive dimensions.
     // /
    test('should calculate the area of a rectangle with positive dimensions', () => {
        expect(calculateRectangleArea(5, 10)).toBe(50);
    });

    /** */
     // Test case: Calculates the area of a rectangle with one dimension being 0.
     // /
    test('should calculate the area of a rectangle with one dimension being 0', () => {
        expect(calculateRectangleArea(0, 10)).toBe(0);
    });

    /** */
     // Test case: Calculates the area of a rectangle with both dimensions being 0.
     // /
    test('should calculate the area of a rectangle with both dimensions being 0', () => {
        expect(calculateRectangleArea(0, 0)).toBe(0);
    });

    /** */
     // Test case: Throws an error when the length is negative.
     // /
    test('should throw an error when the length is negative', () => {
        expect(() => calculateRectangleArea(-5, 10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when the width is negative.
     // /
    test('should throw an error when the width is negative', () => {
        expect(() => calculateRectangleArea(5, -10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when both dimensions are negative.
     // /
    test('should throw an error when both dimensions are negative', () => {
        expect(() => calculateRectangleArea(-5, -10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when the length is not a number.
     // /
    test('should throw an error when the length is not a number', () => {
        expect(() => calculateRectangleArea('abc', 10)).toThrow("Both length and width must be numbers");
    });

    /** */
     // Test case: Throws an error when the width is not a number.
     // /
    test('should throw an error when the width is not a number', () => {
        expect(() => calculateRectangleArea(5, 'abc')).toThrow("Both length and width must be numbers");
    });

    /** */
     // Test case: Throws an error when both dimensions are not numbers.
     // /
    test('should throw an error when both dimensions are not numbers', () => {
        expect(() => calculateRectangleArea('abc', 'def')).toThrow("Both length and width must be numbers");
    });

    /** */
     // Test case: Calculates the area of a rectangle with decimal dimensions.
     // /
    test('should calculate the area of a rectangle with decimal dimensions', () => {
        expect(calculateRectangleArea(2.5, 3.5)).toBe(8.75);
    });
});

describe('calculateTriangleArea', () => {
    /** */
     // Test case: Calculates the area of a triangle with positive dimensions.
     // /
    test('should calculate the area of a triangle with positive dimensions', () => {
        expect(calculateTriangleArea(5, 10)).toBe(25);
    });

    /** */
     // Test case: Calculates the area of a triangle with one dimension being 0.
     // /
    test('should calculate the area of a triangle with one dimension being 0', () => {
        expect(calculateTriangleArea(0, 10)).toBe(0);
    });

    /** */
     // Test case: Calculates the area of a triangle with both dimensions being 0.
     // /
    test('should calculate the area of a triangle with both dimensions being 0', () => {
        expect(calculateTriangleArea(0, 0)).toBe(0);
    });

    /** */
     // Test case: Throws an error when the base is negative.
     // /
    test('should throw an error when the base is negative', () => {
        expect(() => calculateTriangleArea(-5, 10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when the height is negative.
     // /
    test('should throw an error when the height is negative', () => {
        expect(() => calculateTriangleArea(5, -10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when both dimensions are negative.
     // /
    test('should throw an error when both dimensions are negative', () => {
        expect(() => calculateTriangleArea(-5, -10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when the base is not a number.
     // /
    test('should throw an error when the base is not a number', () => {
        expect(() => calculateTriangleArea('abc', 10)).toThrow("Both base and height must be numbers");
    });

    /** */
     // Test case: Throws an error when the height is not a number.
     // /
    test('should throw an error when the height is not a number', () => {
        expect(() => calculateTriangleArea(5, 'abc')).toThrow("Both base and height must be numbers");
    });

    /** */
     // Test case: Throws an error when both dimensions are not numbers.
     // /
    test('should throw an error when both dimensions are not numbers', () => {
        expect(() => calculateTriangleArea('abc', 'def')).toThrow("Both base and height must be numbers");
    });

    /** */
     // Test case: Calculates the area of a triangle with decimal dimensions.
     // /
    test('should calculate the area of a triangle with decimal dimensions', () => {
        expect(calculateTriangleArea(2.5, 3.5)).toBe(4.375);
    });
});

describe('calculateCirclePerimeter', () => {
    /** */
     // Test case: Calculates the perimeter of a circle with a positive radius.
     // /
    test('should calculate the perimeter of a circle with a positive radius', () => {
        expect(calculateCirclePerimeter(5)).toBeCloseTo(2 * Math.PI * 5);
    });

    /** */
     // Test case: Calculates the perimeter of a circle with a radius of 0.
     // /
    test('should calculate the perimeter of a circle with a radius of 0', () => {
        expect(calculateCirclePerimeter(0)).toBe(0);
    });

    /** */
     // Test case: Throws an error when the radius is negative.
     // /
    test('should throw an error when the radius is negative', () => {
        expect(() => calculateCirclePerimeter(-5)).toThrow("Radius cannot be negative");
    });

    /** */
     // Test case: Throws an error when the radius is not a number.
     // /
    test('should throw an error when the radius is not a number', () => {
        expect(() => calculateCirclePerimeter('abc')).toThrow("Radius must be a number");
    });

    /** */
     // Test case: Throws an error when the radius is NaN.
     // /
    test('should throw an error when the radius is NaN', () => {
        expect(() => calculateCirclePerimeter(NaN)).toThrow("Radius must be a number");
    });

    /** */
     // Test case: Calculates the perimeter of a circle with a large radius.
     // /
    test('should calculate the perimeter of a circle with a large radius', () => {
        expect(calculateCirclePerimeter(1000)).toBeCloseTo(2 * Math.PI * 1000);
    });

    /** */
     // Test case: Calculates the perimeter of a circle with a decimal radius.
     // /
    test('should calculate the perimeter of a circle with a decimal radius', () => {
        expect(calculateCirclePerimeter(2.5)).toBeCloseTo(2 * Math.PI * 2.5);
    });
});

describe('calculateRectanglePerimeter', () => {
    /** */
     // Test case: Calculates the perimeter of a rectangle with positive dimensions.
     // /
    test('should calculate the perimeter of a rectangle with positive dimensions', () => {
        expect(calculateRectanglePerimeter(5, 10)).toBe(30);
    });

    /** */
     // Test case: Calculates the perimeter of a rectangle with one dimension being 0.
     // /
    test('should calculate the perimeter of a rectangle with one dimension being 0', () => {
        expect(calculateRectanglePerimeter(0, 10)).toBe(20);
    });

    /** */
     // Test case: Calculates the perimeter of a rectangle with both dimensions being 0.
     // /
    test('should calculate the perimeter of a rectangle with both dimensions being 0', () => {
        expect(calculateRectanglePerimeter(0, 0)).toBe(0);
    });

    /** */
     // Test case: Throws an error when the length is negative.
     // /
    test('should throw an error when the length is negative', () => {
        expect(() => calculateRectanglePerimeter(-5, 10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when the width is negative.
     // /
    test('should throw an error when the width is negative', () => {
        expect(() => calculateRectanglePerimeter(5, -10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when both dimensions are negative.
     // /
    test('should throw an error when both dimensions are negative', () => {
        expect(() => calculateRectanglePerimeter(-5, -10)).toThrow("Dimensions cannot be negative");
    });

    /** */
     // Test case: Throws an error when the length is not a number.
     // /
    test('should throw an error when the length is not a number', () => {
        expect(() => calculateRectanglePerimeter('abc', 10)).toThrow("Both length and width must be numbers");
    });

    /** */
     // Test case: Throws an error when the width is not a number.
     // /
    test('should throw an error when the width is not a number', () => {
        expect(() => calculateRectanglePerimeter(5, 'abc')).toThrow("Both length and width must be numbers");
    });

    /** */
     // Test case: Throws an error when both dimensions are not numbers.
     // /
    test('should throw an error when both dimensions are not numbers', () => {
        expect(() => calculateRectanglePerimeter('abc', 'def')).toThrow("Both length and width must be numbers");
    });

    /** */
     // Test case: Calculates the perimeter of a rectangle with decimal dimensions.
     // /
    test('should calculate the perimeter of a rectangle with decimal dimensions', () => {
        expect(calculateRectanglePerimeter(2.5, 3.5)).toBe(12);
    });
});

describe('calculateDistance', () => {
    /** */
     // Test case: Calculates the distance between two points.
     // /
    test('should calculate the distance between two points', () => {
        expect(calculateDistance(0, 0, 3, 4)).toBe(5);
    });

    /** */
     // Test case: Calculates the distance between two identical points.
     // /
    test('should calculate the distance between two identical points', () => {
        expect(calculateDistance(0, 0, 0, 0)).toBe(0);
    });

    /** */
     // Test case: Calculates the distance between two points with negative coordinates.
     // /
    test('should calculate the distance between two points with negative coordinates', () => {
        expect(calculateDistance(-1, -1, -4, -5)).toBe(5);
    });

    /** */
     // Test case: Throws an error when x1 is not a number.
     // /
    test('should throw an error when x1 is not a number', () => {
        expect(() => calculateDistance('abc', 0, 0, 0)).toThrow("All coordinates must be numbers");
    });

    /** */
     // Test case: Throws an error when y1 is not a number.
     // /
    test('should throw an error when y1 is not a number', () => {
        expect(() => calculateDistance(0, 'abc', 0, 0)).toThrow("All coordinates must be numbers");
    });

    /** */
     // Test case: Throws an error when x2 is not a number.
     // /
    test('should throw an error when x2 is not a number', () => {
        expect(() => calculateDistance(0, 0, 'abc', 0)).toThrow("All coordinates must be numbers");
    });

    /** */
     // Test case: Throws an error when y2 is not a number.
     // /
    test('should throw an error when y2 is not a number', () => {
        expect(() => calculateDistance(0, 0, 0, 'abc')).toThrow("All coordinates must be numbers");
    });

    /** */
     // Test case: Calculates the distance between two points with decimal coordinates.
     // /
    test('should calculate the distance between two points with decimal coordinates', () => {
        expect(calculateDistance(0.5, 0.5, 3.5, 4.5)).toBeCloseTo(5);
    });

    /** */
     // Test case: Calculates distance with large coordinate values.
     // /
    test('should calculate distance with large coordinate values', () => {
        expect(calculateDistance(1000, 1000, 2000, 2000)).toBeCloseTo(1414.21356);
    });
});

File: testcases/vertex_ai/GeometryTest_3c3c033f.java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class GeometryTest {

    private Geometry geometry;

    @BeforeEach
    void setUp() {
        geometry = new Geometry();
    }

    @Test
    void calculateCircleArea_positiveRadius_returnsCorrectArea() {
        double radius = 5.0;
        double expectedArea = Math.PI * radius * radius;
        assertEquals(expectedArea, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateCircleArea_zeroRadius_returnsZeroArea() {
        double radius = 0.0;
        assertEquals(0.0, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateCircleArea_largeRadius_returnsCorrectArea() {
        double radius = 100.0;
        double expectedArea = Math.PI * radius * radius;
        assertEquals(expectedArea, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateCircleArea_negativeRadius_throwsIllegalArgumentException() {
        double radius = -1.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateCircleArea(radius);
        });
        assertEquals("Radius cannot be negative", exception.getMessage());
    }

    @Test
    void calculateCircleArea_smallPositiveRadius_returnsCorrectArea() {
        double radius = 0.1;
        double expectedArea = Math.PI * radius * radius;
        assertEquals(expectedArea, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateRectangleArea_positiveLengthAndWidth_returnsCorrectArea() {
        double length = 5.0;
        double width = 10.0;
        assertEquals(50.0, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateRectangleArea_zeroLength_returnsZeroArea() {
        double length = 0.0;
        double width = 10.0;
        assertEquals(0.0, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateRectangleArea_zeroWidth_returnsZeroArea() {
        double length = 5.0;
        double width = 0.0;
        assertEquals(0.0, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateRectangleArea_negativeLength_throwsIllegalArgumentException() {
        double length = -1.0;
        double width = 10.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateRectangleArea(length, width);
        });
        assertEquals("Dimensions cannot be negative", exception.getMessage());
    }

    @Test
    void calculateRectangleArea_negativeWidth_throwsIllegalArgumentException() {
        double length = 5.0;
        double width = -1.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateRectangleArea(length, width);
        });
        assertEquals("Dimensions cannot be negative", exception.getMessage());
    }

    @Test
    void calculateRectangleArea_largeDimensions_returnsCorrectArea() {
        double length = 100.0;
        double width = 200.0;
        assertEquals(20000.0, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateTriangleArea_positiveBaseAndHeight_returnsCorrectArea() {
        double base = 5.0;
        double height = 10.0;
        assertEquals(25.0, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateTriangleArea_zeroBase_returnsZeroArea() {
        double base = 0.0;
        double height = 10.0;
        assertEquals(0.0, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateTriangleArea_zeroHeight_returnsZeroArea() {
        double base = 5.0;
        double height = 0.0;
        assertEquals(0.0, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateTriangleArea_negativeBase_throwsIllegalArgumentException() {
        double base = -1.0;
        double height = 10.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateTriangleArea(base, height);
        });
        assertEquals("Dimensions cannot be negative", exception.getMessage());
    }

    @Test
    void calculateTriangleArea_negativeHeight_throwsIllegalArgumentException() {
        double base = 5.0;
        double height = -1.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateTriangleArea(base, height);
        });
        assertEquals("Dimensions cannot be negative", exception.getMessage());
    }

    @Test
    void calculateTriangleArea_largeDimensions_returnsCorrectArea() {
        double base = 100.0;
        double height = 200.0;
        assertEquals(10000.0, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateCirclePerimeter_positiveRadius_returnsCorrectPerimeter() {
        double radius = 5.0;
        double expectedPerimeter = 2 * Math.PI * radius;
        assertEquals(expectedPerimeter, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateCirclePerimeter_zeroRadius_returnsZeroPerimeter() {
        double radius = 0.0;
        assertEquals(0.0, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateCirclePerimeter_largeRadius_returnsCorrectPerimeter() {
        double radius = 100.0;
        double expectedPerimeter = 2 * Math.PI * radius;
        assertEquals(expectedPerimeter, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateCirclePerimeter_negativeRadius_throwsIllegalArgumentException() {
        double radius = -1.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateCirclePerimeter(radius);
        });
        assertEquals("Radius cannot be negative", exception.getMessage());
    }

     @Test
    void calculateCirclePerimeter_smallPositiveRadius_returnsCorrectPerimeter() {
        double radius = 0.1;
        double expectedPerimeter = 2 * Math.PI * radius;
        assertEquals(expectedPerimeter, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_positiveLengthAndWidth_returnsCorrectPerimeter() {
        double length = 5.0;
        double width = 10.0;
        assertEquals(30.0, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_zeroLength_returnsCorrectPerimeter() {
        double length = 0.0;
        double width = 10.0;
        assertEquals(20.0, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_zeroWidth_returnsCorrectPerimeter() {
        double length = 5.0;
        double width = 0.0;
        assertEquals(10.0, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_negativeLength_throwsIllegalArgumentException() {
        double length = -1.0;
        double width = 10.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateRectanglePerimeter(length, width);
        });
        assertEquals("Dimensions cannot be negative", exception.getMessage());
    }

    @Test
    void calculateRectanglePerimeter_negativeWidth_throwsIllegalArgumentException() {
        double length = 5.0;
        double width = -1.0;
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
            geometry.calculateRectanglePerimeter(length, width);
        });
        assertEquals("Dimensions cannot be negative", exception.getMessage());
    }

    @Test
    void calculateRectanglePerimeter_largeDimensions_returnsCorrectPerimeter() {
        double length = 100.0;
        double width = 200.0;
        assertEquals(600.0, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateDistance_positiveCoordinates_returnsCorrectDistance() {
        double x1 = 1.0;
        double y1 = 2.0;
        double x2 = 4.0;
        double y2 = 6.0;
        double expectedDistance = 5.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_zeroCoordinates_returnsCorrectDistance() {
        double x1 = 0.0;
        double y1 = 0.0;
        double x2 = 0.0;
        double y2 = 0.0;
        assertEquals(0.0, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_negativeCoordinates_returnsCorrectDistance() {
        double x1 = -1.0;
        double y1 = -2.0;
        double x2 = -4.0;
        double y2 = -6.0;
        double expectedDistance = 5.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_mixedCoordinates_returnsCorrectDistance() {
        double x1 = -1.0;
        double y1 = 2.0;
        double x2 = 4.0;
        double y2 = -6.0;
        double expectedDistance = Math.sqrt(89);
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_largeCoordinates_returnsCorrectDistance() {
        double x1 = 100.0;
        double y1 = 200.0;
        double x2 = 400.0;
        double y2 = 600.0;
        double expectedDistance = 500.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }
}